src/commands/gmotte/bypass-perm/generate.ts

```diff
@@ -0,0 +1,287 @@
+          const bpName = getByPassCustomPermissionName(sobject, automation);
+          if (this.retrievedComponentSet.find((component) => component.fullName === bpName)) return false;
+          return true;
+        });
+        if (automationsForSObject.length > 0) acc[sobject] = automationsForSObject;
+        return acc;
+      },
+      {},
+    );
+  }
+
+  protected async generateCustomPermissions(): Promise<void> {
+    // TODO prompt for output dir
+    // default : default packagedir, if it exists, ./main/default
+    const builder = new xml2js.Builder();
+    const promises: Promise<void>[] = [];
+
+    const outputdir = await this.getDefaultCustomMetadataDir();
+
+    for (const [sobject, automations] of Object.entries(this.bypassCustomPermissionsToGenerate)) {
+      automations.forEach((automation) => {
+        const label = getByPassCustomPermissionName(sobject, automation);
+        const customPermission: AnyJson = {
+          CustomPermission: {
+            $: { xmlns: 'http://soap.sforce.com/2006/04/metadata' },
+            isLicensed: false,
+            label,
+          },
+        };
+        const xmlFile = builder.buildObject(customPermission);
+        const outputFilePath = path.join(outputdir, `${label}.customPermission-meta.xml`);
+        promises.push(fs.writeFile(outputFilePath, xmlFile, 'utf8'));
+      });
+    }
+    await Promise.all(promises);
+  }
+
+  protected async getCustomPermissionsToDeploy(): Promise<ComponentSet> {
+    const allCustomPermissionsComponentSet = await ComponentSetBuilder.build({
+      apiversion: this.getFlag<string>('apiversion'),
+      sourceapiversion: await this.getSourceApiVersion(),
+      metadata: {
+        metadataEntries: ['CustomPermission'],
+        directoryPaths: [this.project.getDefaultPackage().fullPath],
+      },
+    });
+
+    return allCustomPermissionsComponentSet.filter((customPermission) => {
+      const isByPassCustomPermission = isByPassCustomPermissionName(customPermission.fullName);
+      const alreadyExists = !!this.retrieveResult.components.find(
+        (retrievedCustomPermission) => retrievedCustomPermission.fullName === customPermission.fullName,
+      );
+      return isByPassCustomPermission && !alreadyExists;
+    });
+  }
+
+  protected async generateManifest(): Promise<void> {
+    const customPermissionsToAddToPackage = await this.getCustomPermissionsToDeploy();
+    if (customPermissionsToAddToPackage.size > 0) {
+      await fs.writeFile(
+        path.join(await this.getDefaultManifestDir(), packageXmlName),
+        await customPermissionsToAddToPackage.getPackageXml(),
+      );
+    } else {
+      await fs.rm(path.join(await this.getDefaultManifestDir(), packageXmlName));
+    }
+  }
+
+  protected async outputResult() {
+    // suggest to deploy or push depending of org.isScratch
+    if ((await this.getCustomPermissionsToDeploy()).size > 0) {
+      if (this.org.isScratch()) {
```

Comment:
```suggestion
      if (this.org.tracksSource()) {
```
Since sandboxes may also have source tracking enabled